src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +104,64 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
         commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def runs = project.nunit.getTestInputAsList(getWhereFilters())
```

Comment:
previously the dependency on nunit was restricted to OpenCoverNUnitPlugin (applying nunit conventions on top of a non-nunit-dependent opencover plugin).
This design should probably be kept, don;t you think ?

Reply:
I thought about that, but it would mean that some functions, like the one which splits the where conditions should be duplicated here, right?
So, basically in order to keep the current dependency we shouldn't have in the OpenCover.groovy file references to project.nunit stuff, right?

Reply:
yes, though I wouldn't expose the where condition & stuff here, maybe instead you could add something like a field parallelTargetExecArgs (collection of the arguments for each run), and map that as a convention in opencover-nunit plugin.

Reply:
Done